jobs/sql/extractions/functions/temp_program.sql

```diff
@@ -200,15 +200,16 @@ END
 
 DROP FUNCTION IF EXISTS temp_program_encounter_count;
 #
-CREATE FUNCTION temp_program_encounter_count(_patient_program_id int, _encounter_type_id int)
+CREATE FUNCTION temp_program_encounter_count(_patient_program_id int, _encounter_type_id int, _end_datetime datetime)
     RETURNS int
     DETERMINISTIC
 BEGIN
     DECLARE ret int;
     select      count(e.encounter_id) into ret
     from        temp_program_encounter e
     where       e.patient_program_id = _patient_program_id
-      and       e.encounter_type_id = _encounter_type_id;
+      and       e.encounter_type_id = _encounter_type_id
+     and        e.encounter_datetime < _end_datetime;
```

Comment:
Maybe allow for _end_datetime to be null?  By adding a check like `and (_enddatetime is null or e.encounter_datetime < _end_datetime).

Also, are you sure that _end_datetime should be non-inclusive?